lazyproto/def.go

```diff
@@ -0,0 +1,42 @@
+package lazyproto
+
+// NewDef initializes and returns a new Def with mappings for the specified field tags.
+func NewDef(tags ...int) *Def {
+	d := Def{
+		m: map[int]any{},
+	}
+	for _, t := range tags {
+		d.m[t] = nil
+	}
+	return &d
+}
+
+// A Def is an optionally nested mapping of protobuf field tags declaring which values should
+// be decoded from a message.  If the value given tag maps to a nested definition and the wire type
+// in the message data is WireTypeLengthDelimited, the contents are treated as a nested message and
+// the nested mapping is applied to the field data recursively.
+type Def struct {
+	m map[int]any
+}
+
+// Add adds field tags to the mapping and returns the Def
+func (d *Def) Add(tags ...int) *Def {
+	for _, t := range tags {
+		d.m[t] = nil
+	}
+	return d
+}
+
+// AddNested adds a mapping for tag to a nested Def with the specified field tags for the nested message
+// and returns the nested Def
+func (d *Def) AddNested(tag int, nestedTags ...int) *Def {
+	nd := NewDef(nestedTags...)
+	d.m[tag] = nd
```

Comment:
this seem slike it would add to an existing key, but its more like a Set, where it overwrites the existing. Is this meant to add the new nested tags to the existing ones?

Reply:
It does overwrite any existing mapping for that key.  The intent is for these methods to be called just once as part of a builder pattern.  `Add`/`AddNested` felt better than `Set`/`SetNested` to me.

Reply:
I think it could throw folks. When you use the header type and you'll see `Add` and `Set` with different semantics and functionality. Similar to how they set to overwrite in the header, I'd think you'd want to use Set here as well. Add to me says extend, Set says replace... food for thought 🤷

Reply:
I renamed `Add()` to `Tags()` and `AddNested()` to `NestedTag()`.  I think those names read clearer.  I also called out the overwrite/replace behavior in the doc comments for both methods.